svid/jwtsvid/tests/check_svid.c

```diff
@@ -144,6 +180,136 @@ START_TEST(test_jwtsvid_ParseAndValidate)
+          "BEb2UiLCJpYXQiOjE1MTYyMzkwMjIsImV4cCI6MTYyMDAwMDAwMH0.WA7x2GiNZvh5BoLk"
+          "vS7BBGIHz6ULTCsX7DBJo8kDoPla4wbo4G2157WWCZLx6zPE8Qpvvb11kMk0Ivk_G0gMeA";
+    jwtbundle_Source *source = jwtbundle_SourceFromBundle(bundle);
+    jwtsvid_SVID *svid = jwtsvid_ParseAndValidate(token, source, NULL, &err);
+
+    ck_assert_uint_eq(err, NO_ERROR);
+    ck_assert_ptr_eq(svid->audience, NULL);
+    ck_assert_ptr_ne(svid->claims, NULL);
+    ck_assert_uint_eq(shlenu(svid->claims), 4);
+    ck_assert_int_ge(shgeti(svid->claims, "sub"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "name"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "iat"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "exp"), 0);
+    ck_assert_int_eq(svid->expiry, 1620000000);
+    ck_assert_ptr_ne(svid->id.path, NULL);
+    ck_assert_str_eq(svid->id.path, "/workload1");
+    ck_assert_ptr_ne(svid->id.td.name, NULL);
+    ck_assert_str_eq(svid->id.td.name, "example.com");
+    ck_assert_ptr_ne(svid->token, NULL);
+    ck_assert_str_eq(svid->token, token);
+
+    jwtbundle_Source_Free(source);
+    EVP_PKEY_free(pkey);
+    jwtsvid_SVID_Free(svid);
+
+}
+END_TEST
+
+// precondition: valid elliptic curve jwt token
+// postcondition:  valid jwt svid corresponding to the token with valid
+START_TEST(test_jwtsvid_Marshal)
+{
+    spiffeid_TrustDomain td = { "example.com" };
+    jwtbundle_Bundle *bundle = jwtbundle_New(td);
+
+    FILE *f = fopen("./resources/ec-secp256k1-priv-key.pem", "r");
+    ck_assert_ptr_ne(f, NULL);
+    EVP_PKEY *pkey = PEM_read_PrivateKey(f, NULL, NULL, NULL);
+    fclose(f);
+
+    err_t err = jwtbundle_Bundle_AddJWTAuthority(
+        bundle, "ff3c5c96-392e-46ef-a839-6ff16027af78", pkey);
+
+    ck_assert_uint_eq(err, NO_ERROR);
+
+    char token[]
+        = "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImZmM2M1Yzk2LTM5MmUtNDZlZi1hODM5LTZmZjE2MDI3YWY3OCJ9.eyJzdWIiOiJzcGlmZmU6Ly9leGFtcGxlLmNvbS93b3JrbG9hZDEiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjIsImV4cCI6MTYyMDAwMDAwMH0.WA7x2GiNZvh5BoLkvS7BBGIHz6ULTCsX7DBJo8kDoPla4wbo4G2157WWCZLx6zPE8Qpvvb11kMk0Ivk_G0gMeA";
+    jwtbundle_Source *source = jwtbundle_SourceFromBundle(bundle);
+    jwtsvid_SVID *svid = jwtsvid_ParseAndValidate(token, source, NULL, &err);
+
+    ck_assert_uint_eq(err, NO_ERROR);
+    ck_assert_ptr_eq(svid->audience, NULL);
+    ck_assert_ptr_ne(svid->claims, NULL);
+    ck_assert_uint_eq(shlenu(svid->claims), 4);
+    ck_assert_int_ge(shgeti(svid->claims, "sub"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "name"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "iat"), 0);
+    ck_assert_int_ge(shgeti(svid->claims, "exp"), 0);
+    ck_assert_int_eq(svid->expiry, 1620000000);
+    ck_assert_ptr_ne(svid->id.path, NULL);
+    ck_assert_str_eq(svid->id.path, "/workload1");
+    ck_assert_ptr_ne(svid->id.td.name, NULL);
+    ck_assert_str_eq(svid->id.td.name, "example.com");
+    ck_assert_ptr_ne(svid->token, NULL);
+    ck_assert_str_eq(svid->token, token);
+
```

Comment:
This block of code looks like a copy of the code from lines 210 to 240. Couldn't refactor to a function and reuse it?
If you do so, please use a meaningful name for the function.